Dapper.Contrib/SqlMapperExtensions.cs

```diff
@@ -202,7 +202,16 @@ private static PropertyInfo GetSingleKey<T>(string method)
                 foreach (var property in TypePropertiesCache(type))
                 {
                     var val = res[property.Name];
-                    property.SetValue(obj, Convert.ChangeType(val, property.PropertyType), null);
+                    if (val == null) continue;
+                    if (property.PropertyType.IsGenericType() && property.PropertyType.GetGenericTypeDefinition() == typeof(Nullable<>))
+                    {
+                        var genericType = property.PropertyType.GetGenericArguments()[0];

```

Comment:
Since .GetGenericArguments() allocates an array every time, instead use: `Nullable.GetUnderlyingType(property.PropertyType)` to do this much cheaper. Same below :)